tezos/new_context/src/hash/mod.rs

```diff
@@ -260,35 +297,49 @@ pub(crate) fn hash_blob(blob: &ContextValue) -> Result<EntryHash, HashingError>
 // <time in epoch format (8bytes)
 // <commit author name length (8bytes)><commit author name bytes>
 // <commit message length (8bytes)><commit message bytes>
-pub(crate) fn hash_commit(commit: &Commit) -> Result<EntryHash, HashingError> {
+pub(crate) fn hash_commit(
+    commit: &Commit,
+    store: &mut ContextKeyValueStore,
+) -> Result<HashId, HashingError> {
     let mut hasher = VarBlake2b::new(ENTRY_HASH_LEN)?;
     hasher.update(&(ENTRY_HASH_LEN as u64).to_be_bytes());
-    hasher.update(&commit.root_hash);
+
+    let root_hash = store
+        .get_hash(commit.root_hash)?
+        .ok_or(HashingError::ValueExpected("root_hash"))?;
+    hasher.update(root_hash.as_ref());
 
     if commit.parent_commit_hash.is_none() {
         hasher.update(&(0_u64).to_be_bytes());
     } else {
-        let parent_commit_hash = commit
-            .parent_commit_hash
+        let parent_commit_hash = store
+            .get_hash(commit.parent_commit_hash.unwrap())?
```

Comment:
@sebastiencs a new `unwrap()` was introduced here, but it should `.ok_or(HashingError::ValueExpected("parent_commit_hash"))?` as before.

Reply:
@tizoc This statement is actually in an `else` block of the condition `commit.parent_commit_hash.is_none()`.
So `commit.parent_commit_hash` is never `None`.
I will change the code to make that more clear